src/sshping.cxx

```diff
@@ -90,41 +99,158 @@ enum  { opNONE,
+    {opREM,  0, "z", "remote",        Arg::Reqd,     "  -z  --remote FILE    Remote file for up/download tests;"},
+    {opNONE, 0, "",  "",              Arg::None,     "                           default=/tmp/sshping-PID.tmp" },
     {0,0,0,0,0,0}
 };
 /* *INDENT-ON* */
 
+#ifdef _WIN32
+  #include <Ws2tcpip.h>
+  #include <windows.h>
+  #include <conio.h>
+
+  double PCFreq = 0;
+
+// Replacement for the getpass UNIX method
+  char *getpass(const char *prompt) {
+	  static const int PASS_MAX = 512;
+      char getpassbuf[PASS_MAX + 1];
+      size_t i = 0;
+      int c;
+      if (prompt) {
+          fputs(prompt, stderr);
+          fflush(stderr);
+      }
+      for (;;) {
+      	c = _getch();
+          if (c != 0) {
+              if (c == '\r') {
+                  getpassbuf[i] = '\0';
+                  break;
+              }
+              else if (c == '\3') {
+                  exit(0);
+              }
+              else if (c == '\b' && i != 0) {
+                  getpassbuf[i] = NULL;
+                  i--;
+              }
+              else if (i < PASS_MAX && c != '\b') {
+                  getpassbuf[i++] = c;
+              }
+              if (i >= PASS_MAX) {
+                  getpassbuf[i] = '\0';
+                  break;
+              }
+      	  }
+      }
+      if (prompt) {
+          fputs("\r\n", stderr);
+          fflush(stderr);
+      }
+      return _strdup(getpassbuf);
+  }
+
+// Replacement for the getpid UNIX method
+  DWORD getpid() {
+      return GetCurrentProcessId();
+  }
+
+// Replacement for the strsep UNIX method
+  char* strsep(char** stringp, const char* delim) {
+      char* start = *stringp;
+      char* p;
+      p = (start != NULL) ? strpbrk(start, delim) : NULL;
+      if (p == NULL) {
+          *stringp = NULL;
+      }
+      else {
+          *p = '\0';
+          *stringp = p + 1;
+      }
+      return start;
+  } 
+
+// Replacement for the clock_gettime UNIX method
+  uint64_t get_time() {
+      LARGE_INTEGER li;
+      long temp = 0;
+      if (PCFreq == 0) {
+          QueryPerformanceFrequency(&li);
+          PCFreq = (double)li.QuadPart / GIGA;
+      }
+      QueryPerformanceCounter(&li);
+      return static_cast<uint64_t>(static_cast<double>(li.QuadPart) / PCFreq);
+  }
+
+  void keyboard_wait() {
+      _getch();
+  }
+#else
+  #include <termios.h>
+  #include <arpa/inet.h>
+  void keyboard_wait() {
+      static struct termios oldt, newt;
+      tcgetattr(0, &oldt);
+      newt = oldt;
+      newt.c_lflag &= ~ICANON;
+      newt.c_lflag &= ~ECHO;
+      tcsetattr(0, TCSANOW, &newt);
+      ch = getchar();
+      tcsetattr(0, TCSANOW, &oldt);
+  }
+  uint64_t get_time() {
+      struct timespec tz;
+      clock_gettime(CLOCK_MONOTONIC, &tz);
+      uint64_t output = (tz.tv_sec * GIGA + tz.tv_nsec);
+      return output;
+  }
+#endif
+
 // Outta here!
-void die(const char* msg) {
+void die(const char* msg, int exit_no) {
```

Comment:
239 - 254

These lines could be simplified as:

```
void die(const int exit_no) {
    if (key_wait) {
        printf("Press any key to exit...\n");
        keyboard_wait();
    }
    exit(exit_no);
}

void die(const char* msg, const int exit_no) {
    fprintf(stderr, "*** %s\n", msg);
    die(exit_no);
}
```

You have to swap the order because a function must be declared before it is called